plugins/hls-class-plugin/src/Ide/Plugin/Class.hs

```diff
@@ -234,18 +253,55 @@ codeAction state plId (CodeActionParams _ _ docId _ context) = liftIO $ fmap (fr
           _ -> panic "Ide.Plugin.Class.findClassFromIdentifier"
     findClassFromIdentifier _ (Left _) = panic "Ide.Plugin.Class.findClassIdentifier"
 
+    findImplementedMethods :: NormalizedFilePath -> Range -> MaybeT IO [T.Text]
+    findImplementedMethods docPath range = do
+        (HAR {hieAst = hf}, pmap) <- MaybeT . runAction "classplugin" state $ useWithStale GetHieAst docPath
+        instanceRange <- MaybeT . pure $ fromCurrentRange pmap range
+        pure
+            $ concat
+            $ pointCommand hf (instanceRange ^. J.start & J.character -~ 1)
+            $ map (T.pack . getOccString) . rights . findInstanceValBindIdentifiers
+
+    -- | Recurses through the given AST to find identifiers which are are
+    -- 'InstanceValBind's. Two types of nodes are chosen to look inside:
+    --
+    -- 1. The nodes which don't have any identifiers: these are the nodes which
+    -- somehow wrap an expression, like the instance declaration itself, or a
+    -- 'Match' which contains the 'InstanceValBind'.
+    --
+    -- 2. The nodes which have an 'EvidenceVarBind': if one of the implemented
+    -- methods uses functions like 'undefined', the binding for the '?callstack'
+    -- gets bound over the 'Match' which contains the 'InstanceValBind', so to
+    -- find the 'InstanceValBind' this function must look inside any
+    -- 'EvidenceVarBind's it finds.
+    findInstanceValBindIdentifiers :: HieAST a -> [Identifier]
+    findInstanceValBindIdentifiers ast
+      | Map.null nodeIds
+        || hasEvidenceBind nodeIds = concatMap findInstanceValBindIdentifiers (nodeChildren ast)
+      | otherwise = Map.keys
+                  . Map.filter (not . Set.null)
+                  . Map.map (Set.filter isInstanceValBind . identInfo)
```

Comment:
```suggestion
      | otherwise = Map.keys
                  . Map.filter (not . Set.null . Set.filter isInstanceValBind . identInfo)
```

Reply:
I reduced it further to `Map.filter (any isInstanceValBind . identInfo)`